src/features/ajax/aggregate/gql.js

```diff
@@ -0,0 +1,97 @@
+import { isTrueObject } from '../../../common/util/type-check'
+
+/**
+ * @typedef {object} GQLMetadata
+ * @property {string} operationName Name of the operation
+ * @property {string} operationType Type of the operation
+ * @property {string} operationFramework Framework responsible for the operation
+ */
+
+/**
+ *
+ * @param {object|string} body Ajax request body
+ * @param {string} query Ajax request query param string
+ * @returns {GQLMetadata | undefined}
+ */
+export function parseGQL ({ body, query } = {}) {
+  if (!body && !query) return
+  const gqlBody = parseBatchGQL(parseGQLContents(body))
+  if (gqlBody) return gqlBody
+  const gqlQuery = parseSingleGQL(parseGQLQueryString(query))
+  if (gqlQuery) return gqlQuery
+}
+
+/**
+ * @param {string|Object} gql The GraphQL object body sent to a GQL server
+ * @returns {GQLMetadata}
+ */
+function parseSingleGQL (contents) {
+  if (typeof contents !== 'object' || !contents.query || typeof contents.query !== 'string') return
+
+  /** parses gql query string and returns [fullmatch, type match, name match] */
+  const matches = contents.query.trim().match(/^(query|mutation|subscription)\s?(\w*)/)
+  const operationType = matches?.[1]
+  if (!operationType) return
+  const operationName = contents.operationName || matches?.[2] || 'Anonymous'
+  return {
+    operationName, // the operation name of the indiv query
+    operationType, // query, mutation, or subscription,
+    operationFramework: 'GraphQL'
+  }
+}
+
+function parseBatchGQL (contents) {
+  if (!contents) return
+  if (!Array.isArray(contents)) contents = [contents]
+
+  const opNames = []
+  const opTypes = []
+  for (let content of contents) {
+    const operation = parseSingleGQL(content)
+    if (!operation) continue
+
+    opNames.push(operation.operationName)
+    opTypes.push(operation.operationType)
+  }
+
+  if (!opTypes.length) return
+  return {
+    operationName: opNames.join(','), // the operation name of the indiv query -- joined by ',' for batched results
+    operationType: opTypes.join(','), // query, mutation, or subscription -- joined by ',' for batched results
+    operationFramework: 'GraphQL'
```

Comment:
Do we need an additional property here or a different framework name to indicate it was a batched gql call?

Reply:
I think we could add that later if necessary.  Unclear if that will be useful and could increase ingest arbitrarily (same could be said for framework tbh, but rn that seems more "needed" as it is the only thing that indicates what the data is representing)